src/config.rs

```diff
@@ -393,23 +393,103 @@ impl Config {
             Ok(())
         }
 
-        const ID_HEADER_PATTERN: &str = r"(?m)^\s*id\s*:[\sA-Za-z\d]*$";
+        /// returns true when the next occurrence of id header needs to be ignored
+        /// for cases when the section-line refers to a component definition
+        /// or if the section is a youtube type component
+        /// since user can put multiple instances of this component
+        /// in various places in the package referring to the same youtube video
+        /// having same id so ignoring these ids.
+        ///
+        /// in any other case returns false
+        fn ignore_next_id(section_line: &str) -> bool {
+            // Strip out initial '-- ' or '--- '
+            let section_line = section_line.trim_start_matches('-').trim_start();
```

Comment:
a section is a group of lines, what is the meaning of `section_line` here

Reply:
this section line refers to the start of the section. I should rename this to `section_start_line`